tests/test_package/test_query_handlers/query_handler_test.py

```diff
@@ -52,34 +60,51 @@ def start(self) -> Union[Continue, Finish[Dict[str, Any]]]:
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[str]]:
         return Finish(result=FINAL_RESULT)
 
 
-class QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext(QueryHandler):
-    def __init__(self, parameter: Dict[str, Any], query_handler_context: ScopeQueryHandlerContext):
+class QueryHandlerTestWithTwoIterationFactory(UDFQueryHandlerFactory):
+
+    def create(self, parameter: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
+        return QueryHandlerTestWithTwoIteration(parameter, query_handler_context)
+
+
+class QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext(UDFQueryHandler):
+    def __init__(self, parameter: str, query_handler_context: ScopeQueryHandlerContext):
         super().__init__(parameter, query_handler_context)
         self.child = None
 
-    def start(self) -> Union[Continue, Finish[Dict[str, Any]]]:
+    def start(self) -> Union[Continue, Finish[str]]:
         self.child = self._query_handler_context.get_child_query_handler_context()
         return Finish(result=FINAL_RESULT)
 
-    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[Dict[str, Any]]]:
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[str]]:
         pass
 
 
-class QueryHandlerWithOneIterationWithNotReleasedTemporaryObject(QueryHandler):
+class QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContextFactory(UDFQueryHandlerFactory):
 
-    def __init__(self, parameter: Dict[str, Any], query_handler_context: ScopeQueryHandlerContext):
+    def create(self, parameter: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
+        return QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext(parameter, query_handler_context)
+
+
+class QueryHandlerWithOneIterationWithNotReleasedTemporaryObject(UDFQueryHandler):
+
+    def __init__(self, parameter: str, query_handler_context: ScopeQueryHandlerContext):
         super().__init__(parameter, query_handler_context)
         self.proxy = None
         self.child = None
 
-    def start(self) -> Union[Continue, Finish[Dict[str, Any]]]:
+    def start(self) -> Union[Continue, Finish[str]]:
         self.child = self._query_handler_context.get_child_query_handler_context()
         self.proxy = self.child.get_temporary_table()
         return Finish(result=FINAL_RESULT)
 
-    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[Dict[str, Any]]]:
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[str]]:
         pass
+
+class QueryHandlerWithOneIterationWithNotReleasedTemporaryObjectFactory(UDFQueryHandlerFactory):
+
+    def create(self, parameter: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
+        return QueryHandlerWithOneIterationWithNotReleasedTemporaryObject(parameter, query_handler_context)
```

Comment:
the factories are all the same, only difference is the returned type. looks like a lot of boiler plate code.

```suggestion
class QueryHandlerTestFactory(UDFQueryHandlerFactory):
    def __init__(self, returned_type):
        self._returned_type = returned_type
        
    def create(self, parameter: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
        return self._returned_type(parameter, query_handler_context)
 
 QueryHandlerWithOneIterationWithNotReleasedTemporaryObjectFactory = QueryHandlerTestFactory(QueryHandlerWithOneIterationWithNotReleasedTemporaryObject)
 QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContextFactory = QueryHandlerTestFactory(QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext)
 ...
```

Reply:
The reason is, we inject the factories by name and not by value or reference. They will be created in the UDF from their name.

Reply:
Before we injected the Query handler directly by the class name without the factories, however this didn't allow us to wrap the actual query handler